### PR TITLE
WFLY-7538 additional socket permission for WebSocketTestCase

### PR DESCRIPTION
**Pull request by istudens**
https://issues.jboss.org/browse/WFLY-7538

### CODE REVIEW

**Review by jmesnil — APPROVED** [review]

File: testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/websocket/WebSocketTestCase.java [hunk 1]
```diff
@@ -40,17 +39,16 @@ public static WebArchive deploy() {
                                 new PropertyPermission("node0", "read"),
                                 new PropertyPermission("jboss.http.port", "read"),
                                 // Needed for the serverContainer.connectToServer()
-                                new SocketPermission("*:" + TestSuiteEnvironment.getHttpPort(), "connect,resolve")
+                                new SocketPermission(TestSuiteEnvironment.getServerAddress() + ":" + TestSuiteEnvironment.getHttpPort(), "connect,resolve"),
```
Comment (kwart) [inline testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/websocket/WebSocketTestCase.java]: This is not a correct fix. The deployment doesn't require any permission. We should not run the test within the deployment, the `@RunAsClient` should be rather used. 

Use sth like this:

```java
@RunWith(Arquillian.class)
@RunAsClient
public class WebSocketTestCase {

    @Deployment(testable=false)
    public static WebArchive deploy() {
        return ShrinkWrap.create(WebArchive.class, "websocket.war")
                .addClasses(AnnotatedEndpoint.class)
                .addAsManifestResource(new StringAsset("io.undertow.websockets.jsr.UndertowContainerProvider"),
                        "services/javax.websocket.ContainerProvider");
    }

    @Test
    public void testWebSocket() throws Exception {
        AnnotatedClient endpoint = new AnnotatedClient();
        WebSocketContainer serverContainer = ContainerProvider.getWebSocketContainer();
        try (Session session = serverContainer.connectToServer(
                endpoint,
                new URI("ws", "", TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getHttpPort(),
                        "/websocket/websocket/Stuart", "", ""))) {
            Assert.assertEquals("Hello Stuart", endpoint.getMessage());
        }
    }
}
```

**Comment by istudens** [comment]
That would be a bit different test case then. I guess there was a reason why to run all this on the server side. Anyway, it is almost the same either change it to run as client or keep it running on server and provide permissions it needs when SM is enabled.

**Comment by kwart** [comment]
The assigned permissions can hide problems with permissions for underlying logic. 

IMO it's more proper to make the client out of the server. @stuartwdouglas WDYT?

**Comment by stuartwdouglas** [comment]
I'm not actually sure about this one. In general the correct fix would be to perform the connect in a privileged block in the Undertow code, however if you are using a security manager you may want to restrict the ability to code to create remote connections. 

I think in this case the fix is ok.

**Comment by kwart** [comment]
After discussion with @rhatlapa (responsible for WebSockets testing in EAP), I've sent new PR #9506  which covers both use-cases:
* client in a deployment
* standalone client

**Comment by istudens** [comment]
That doubles time needed for execution, but it is fine with me. Please close this PR and also create a 7.1 PR if https://github.com/wildfly/wildfly/pull/9506 is approved/reviewed.

**Comment by gaol** [comment]
Could anyone please merge this PR or  https://github.com/wildfly/wildfly/pull/9506 for the WebSocketTestCase failure? EAP 7.0.x has this test case enabled, the failure blocks the CI from getting the green light. :smile: Thank you very much!

**Comment by ivassile** [comment]
@stuartwdouglas Can you please approve this PR or the replacement https://github.com/wildfly/wildfly/pull/9506? We need this change for EAP 7.0.5. Thanks!

**Comment by stuartwdouglas** [comment]
https://github.com/wildfly/wildfly/pull/9506 also looks good to me, I have approved it

**Comment by istudens** [comment]
Superseded by https://github.com/wildfly/wildfly/pull/9506

**Pull request closed without merging**